### ISSUE TITLE
BUGs found in production so far

### ISSUE THREAD

**Issue by mattgodbolt** [mattgodbolt/xania#140]
```
$ cat xania.* | grep 'BUG:.*' -oE | sort | uniq -c | sort -n
      1 BUG: Affect_modify: unknown location -4.
      1 BUG: Affect_modify: unknown location 65.
      1 BUG: Dam_message: bad dt -1.
      1 BUG: Fread_char: no match: Info_email
      1 BUG: Fread_char: no match: Info_name
      1 BUG: Fread_char: no match: Info_url
      1 BUG: Mob:3064 couldnt open mobprog file drunk.prg
      2 BUG: Act: null givenRoom with TO_GIVENROOM.
      2 BUG: Couldn't open tip file 'tipfile.txt' for reading
      2 BUG: The MUD has caught a SIGTERM - gracefully shutting down
      3 BUG: one_hit:QUAD_DAM with acid fang fangs nitrius [1006] by nitrius acid acidic cloud
      4 BUG: Fread_char: no match: NewsRead
      4 BUG: Fread_obj: no match.
     15 BUG: OLC: don't forget to fix #RESETS M <arg4>
     20 BUG: Equip_char: already equipped (11).
     20 BUG: Equip_char: already equipped (17).
     20 BUG: Fix_exits: 12029:1 -> 12082:3 -> 12083.
     20 BUG: Fix_exits: 12081:1 -> 12082:3 -> 12083.
     20 BUG: Fix_exits: 12081:3 -> 12081:1 -> 12082.
     20 BUG: Fix_exits: 12082:0 -> 12084:2 -> 12081.
     20 BUG: Fix_exits: 12082:1 -> 12081:3 -> 12081.
     20 BUG: Fix_exits: 12082:2 -> 12082:0 -> 12084.
     20 BUG: Fix_exits: 12082:3 -> 12083:1 -> 12081.
     20 BUG: Fix_exits: 12083:1 -> 12081:3 -> 12081.
     20 BUG: Fix_exits: 12083:3 -> 12082:1 -> 12081.
     20 BUG: Fix_exits: 12084:0 -> 12081:2 -> 12029.
     20 BUG: Fix_exits: 12084:1 -> 12084:3 -> 12081.
     20 BUG: Fix_exits: 12084:2 -> 12081:0 -> 12083.
     20 BUG: Fix_exits: 12084:3 -> 12081:1 -> 12082.
     20 BUG: Fix_exits: 16000:0 -> 16100:2 -> 16117.
     20 BUG: Fix_exits: 16400:1 -> 16401:3 -> 16404.
     20 BUG: Fix_exits: 16401:2 -> 16400:0 -> 16399.
     20 BUG: Fix_exits: 16401:3 -> 16404:1 -> 16407.
     20 BUG: Fix_exits: 16403:0 -> 16401:2 -> 16400.
     20 BUG: Fix_exits: 16414:2 -> 16195:0 -> 16413.
     20 BUG: Fix_exits: 16498:1 -> 16286:3 -> 16284.
     20 BUG: Fix_exits: 1805:0 -> 1806:2 -> 1809.
     20 BUG: Fix_exits: 1806:2 -> 1809:0 -> 1808.
     20 BUG: Fix_exits: 1837:1 -> 1839:3 -> 1838.
     20 BUG: Fix_exits: 27066:4 -> 27068:5 -> 27067.
     20 BUG: Fix_exits: 27068:5 -> 27067:4 -> 27066.
     20 BUG: Fix_exits: 27085:1 -> 27091:3 -> 27089.
     20 BUG: Fix_exits: 27089:1 -> 27092:3 -> 27085.
     20 BUG: Fix_exits: 27091:3 -> 27089:1 -> 27092.
     20 BUG: Fix_exits: 27092:3 -> 27085:1 -> 27091.
     20 BUG: Fix_exits: 30011:0 -> 3001:2 -> 3005.
     20 BUG: Fix_exits: 30011:2 -> 3001:0 -> 3054.
     20 BUG: Fix_exits: 30011:5 -> 3001:4 -> 3700.
     20 BUG: Fix_exits: 30012:0 -> 30011:2 -> 3001.
     20 BUG: Fix_exits: 817:2 -> 818:0 -> 819.
     20 BUG: Fix_exits: 818:2 -> 817:0 -> 801.
     20 BUG: Fix_exits: 838:2 -> 801:0 -> 802.
     48 BUG: Fread_char: no match.
     49 BUG: Dam_message: bad dt 1041.
```

am guessing the no matches are harmless, along with Fix_exits.

**Comment by snellers** [comment]
The fix_exits and equip_char warnings are fixed by 4f48c42c1dc

**Comment by snellers** [comment]
@mattgodbolt  all of these are fixed now with the exception of `Dam_message: bad dt 1041.`. 

One source of those was fixed in #201 but grepping the logs we saw quite a few log lines reporting it again on the evening of Oct 15, but I was unable to reproduce it. Any time damage() gets called with TYPE_UNDEFINED,  along with some other unknown factor, it can happen.  So I I'm committing an improvement to the call it makes to bug() to tell us about the ch, victim and the dam_type argument too, hopefully that sheds some light.
